src/inputs/TreeSelectField/utils.ts

```diff
@@ -14,6 +14,7 @@ export type TreeFieldState<O> = {
   filteredOptions: LeveledOption<O>[];
   selectedKeys: Key[];
   selectedOptions: NestedOption<O>[];
+  selectedOptionsLabels: string[];
```

Comment:
Suggestion: Maybe an add in-source comment that with trees the selected labels could be "p1, p2c3" even though the selected options are "p1, p1c1, p1c2, p2c3", something like that? If I'm following why this is necessary, kinda guessing...